### PR TITLE
Switch to extension registration mechanism

### PR DESCRIPTION
**Pull request by kghbln**
This PR is made in reference to: #38 

This PR addresses or contains:
- Switches to extension registration mechanism
- Obsoletes https://github.com/SemanticMediaWiki/SemanticCompoundQueries/pull/33

This PR includes:
- [ ] Tests (unit/integration)
- [x] CI build passed

Fixes: #38

### CODE REVIEW

**Comment by kghbln** [comment]
@mwjames It took me a while to add a sensible set of test environments but I guess we are cool now. Since all tests passed a squash and merge should be possible. :)

**Comment by kghbln** [comment]
@mwjames As always, thanks a ton!

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+    - env: DB=mysql; MW=REL1_27; SMW=~3.0@dev
       php: 5.6
+    - env: DB=mysql; MW=REL1_28; SMW=~3.0@dev; TYPE=coverage
+      php: 7.0
+    - env: DB=sqlite; MW=REL1_29; SMW=~3.0@dev
+      php: 7.1
+    - env: DB=postgres; MW=REL1_30; SMW=~3.0@dev
+      php: 7.2      
+  allow_failures:
+    - env: DB=mysql; MW=master; SMW=~3.1@dev
```
Comment (kghbln) [inline .travis.yml]: @mwjames Will it make sense to add PHPUnit to these lines like e.g. in SMT? For a lot of extensions we do not add e.g. `SMW=~3.1@dev`. I am unsure when to use it and when not. I guess this is to be able to specify a distinct branch and if we do not master of SMW is used? I this case SMW could probably be dropped from the lines

Comment (mwjames) [reply .travis.yml]: PHPUnit depends on the require PHP version. MW 1.31 comes with a required PHPUnit version which should work with PHP 7.0 but in some cases MW lacks behind and PHP 7.2/7.3 requires a more recent version of PHPUnit.

So, it all depends, and in some cases you get in discussions with WMF about it [0].

[0] https://phabricator.wikimedia.org/T204420

Comment (kghbln) [reply .travis.yml]: Indeed, I remember this one. I just mixed in permissible PHPUnits. Still not sure about the SMW part ...

**Pull request merged**